### PR TITLE
MAT-5735 String reusable component

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-5735](https://jira.cms.gov/browse/MAT-5735)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is into the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/common/string/StringField.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+const StringField = ({ label, fieldValue, canEdit }: StringFieldProps) => {
+  const [field, setField] = useState<string>(fieldValue);
+  return (
+    <div style={{ width: "134px" }}>
+      <FormControl>
+        <InputLabel
+          id="string-field-label"
+          data-testid="string-field-label"
+          required={false}
```
Comment (RohitKandimalla) [inline src/components/common/string/StringField.tsx]: Can we accept required field as a prop and may be default it to "false" ?

File: src/components/common/string/StringField.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+          sx={labelStyle}
+        >
+          {label}
+        </InputLabel>
+        <TextField
+          disabled={!canEdit}
+          label={label}
```
Comment (RohitKandimalla) [inline src/components/common/string/StringField.tsx]: Do we need 2 labels with same name ? Also can you provide screenshot for the implementation. that way its easier to review.

File: src/components/common/string/StringField.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+          placeholder=""
+          id={`string-field-${label}`}
+          data-testid={`string-field-${label}`}
+          inputProps={{
+            "data-testid": `string-field-${label}-input`,
+            "aria-describedby": `string-field-${label}input-helper-text`,
+            required: true,
```
Comment (RohitKandimalla) [inline src/components/common/string/StringField.tsx]: Is the input always required ?

File: src/components/common/string/StringField.tsx (hunk 1 above)
Comment (sb-cecilialiu) [reply src/components/common/string/StringField.tsx]: @RohitKandimalla Thanks Rohit, just did it.

File: src/components/common/string/StringField.tsx (hunk 3 above)
Comment (sb-cecilialiu) [reply src/components/common/string/StringField.tsx]: @RohitKandimalla shouldn't, changed it to use the passed in prop.

File: src/components/common/string/StringField.tsx (hunk 2 above)
Comment (sb-cecilialiu) [reply src/components/common/string/StringField.tsx]: @RohitKandimalla Yes based on the design, it should be there (https://www.figma.com/file/1RPYLI39hfMczAL81ARArX/MADiE-Application?type=design&node-id=9969-361838&mode=design&t=LTibHEo7B0nfMluY-0)

Comment (sb-cecilialiu) [reply src/components/common/string/StringField.tsx]: Pasted Graphic 1

**Review by RohitKandimalla — APPROVED** [review]

**Pull request merged**